dcos/subcommand.py

```diff
@@ -231,19 +280,32 @@ def _install_env(pkg, options):
 
     pkg_dir = _package_dir(pkg.name())
 
-    install_operation = pkg.command_json(options)
-
-    env_dir = os.path.join(pkg_dir,
-                           constants.DCOS_SUBCOMMAND_VIRTUALENV_SUBDIR)
+    resources = pkg.resource_json()
+    binary_cli = _get_cli_binary(resources)
+    env_dir = os.path.join(pkg_dir)
 
-    if 'pip' in install_operation:
-        _install_with_pip(
+    if binary_cli:
+        _install_with_binary(
             pkg.name(),
             env_dir,
-            install_operation['pip'])
+            binary_cli)
     else:
-        raise DCOSException("Installation methods '{}' not supported".format(
-            install_operation.keys()))
+        install_operation = pkg.command_json(options)
```

Comment:
I'll fix this in another pr. Created https://github.com/dcos/dcos-cli/issues/593